src/Forms/GridField/GridFieldDetailForm_ItemRequest.php

```diff
@@ -627,18 +632,30 @@ public function getEditLink($id)
      */
     private function getGridFieldItemAdjacencies(): array
     {
-        $list = $this->getGridField()->getManipulatedList();
-        $paginator = $this->getGridFieldPaginatorState();
-        if (!$paginator) {
+        $paginatorState = $this->getGridFieldPaginatorState();
+        if (!$paginatorState) {
             return [];
         }
-        $currentPage = $paginator->getData('currentPage');
-        $itemsPerPage = $paginator->getData('itemsPerPage');
-
+        $keyStr = '';
+        $data = $this->getGridField()->getState()->toArray();
+        array_walk_recursive($data, function ($v, $k) use (&$keyStr) {
+            if (is_scalar($k) && is_scalar($v)) {
```

Comment:
Checking for scalars here rather than just using serialize() for the whole thing just incase something in there won't serialize properly